### ISSUE TITLE
Alternator CreateTable and DeleteTable sometimes fails with tablets

### ISSUE THREAD

**Issue by nyh** [scylladb/scylladb#16468]
If I hard-code Alternator and test/alternator/run to always use tablets, with
```patch
diff --git a/alternator/executor.cc b/alternator/executor.cc
index c49e74d7c1..884233cb51 100644
--- a/alternator/executor.cc
+++ b/alternator/executor.cc
@@ -4529,9 +4529,13 @@ static lw_shared_ptr<keyspace_metadata> create_keyspace_metadata(std::string_vie
     // If we make this tag a permanent feature, it will get a "system:" prefix -
     // until then we give it the "experimental:" prefix to not commit to it.
     static constexpr auto INITIAL_TABLETS_TAG_KEY = "experimental:initial_tablets";
+#if 0
     if (tags_map.contains(INITIAL_TABLETS_TAG_KEY)) {
         opts.emplace("initial_tablets", tags_map.at(INITIAL_TABLETS_TAG_KEY));
     }
+#else
+    opts.emplace("initial_tablets", "100");
+#endif
 
     return keyspace_metadata::new_keyspace(keyspace_name, "org.apache.cassandra.locator.NetworkTopologyStrategy", std::move(opts), true);
 }

diff --git a/test/cql-pytest/run.py b/test/cql-pytest/run.py
index 2e342c372e..571c7ceabc 100755
--- a/test/cql-pytest/run.py
+++ b/test/cql-pytest/run.py
@@ -311,6 +311,8 @@ def run_scylla_cmd(pid, dir):
         # test/alternator/run.
         '--experimental-features=udf',
         '--experimental-features=keyspace-storage-options',
+        '--experimental-features=consistent-topology-changes',
+        '--experimental-features=tablets',
         '--enable-user-defined-functions', '1',
         # Set up authentication in order to allow testing this module
         # and other modules dependent on it: e.g. service levels
```

And run `test/alternator/run`, I get failures in some cases of `CreateTable`:
> An error occurred (InternalServerError) when calling the CreateTable operation (reached max retries: 0): Internal server error: exceptions::read_failure_exception (Operation failed for system_auth.roles - received 0 responses and 1 failures from 1 CL=LOCAL_ONE.)

And of `DeleteTable`:
> An error occurred (InternalServerError) when calling the DeleteTable operation (reached max retries: 0): Internal server error: raft::stopped_error (Raft instance is stopped, reason: "background error, std::_Nested_exception<raft::state_machine_error> (State machine error at raft/server.cc:1230): std::_Nested_exception<std::runtime_error> (Failed to read tablet metadata): exceptions::read_failure_exception (Operation failed for system.tablets - received 0 responses and 1 failures from 1 CL=ONE.)")

It seems that this partially depends on luck - if I try to run just  `test/alternator/run test_item.py`, I never see any failure. If I try to run `test/alternator/run test_gsi.py test_item.py` sometimes it works correctly, and a few times I got this:

> An error occurred (InternalServerError) when calling the DeleteTable operation (reached max retries: 0): Internal server error: raft::stopped_error (Raft instance is stopped, reason: "background error, std::_Nested_exception<raft::state_machine_error> (State machine error at raft/server.cc:1230): logalloc::bad_alloc (failed to refill emergency reserve of 30 (have 28 free segments))")

and sometimes the more familiar
> An error occurred (InternalServerError) when calling the DeleteTable operation (reached max retries: 0): Internal server error: raft::stopped_error (Raft instance is stopped, reason: "background error, std::_Nested_exception<raft::state_machine_error> (State machine error at raft/server.cc:1230): std::_Nested_exception<std::runtime_error> (Failed to read tablet metadata): exceptions::read_failure_exception (Operation failed for system.tablets - received 0 responses and 1 failures from 1 CL=ONE.)")

Note: These errors do not happen if I run the same tests with the experimental features turned on, but without the "initial_tablets: 100". So the bug has to be tablet-specific, somehow.

**Comment by nyh** [comment]
It seems the `bad_alloc` might be the cause of all these failures (although, sadly, this wasn't obvious in most of them). If in increase the Scylla memory from 1G to 2G (in test/cql-pytest/run.py), the test runs more tests successfully (there are still failures, but these look like different bugs).

Maybe the tablets code is somehow leaking memory? The Alternator tests do not create and delete too many tables, but they do create and delete some, maybe if we're leaking a lot memory per creation it can be causing this? I need to investigate if this can explain this.

**Comment by nyh** [comment]
This issue is probably not specific to Alternator - we get similar problems also in the cql-pytest tests - see https://github.com/scylladb/scylladb/issues/16473.

**Comment by nyh** [comment]
The good news is that if I run the separate test files individually, and not try to run all of them on one Scylla, all the tests that I expected to test pass except the Alternator Streams (CDC) and TTL tests. These are still known holes in the implementation (see https://github.com/scylladb/scylladb/issues/16355).

**Comment by nyh** [comment]
As I explained in https://github.com/scylladb/scylladb/issues/16473#issuecomment-1864601581, it turns out that the bad_alloc problem was caused by my silly choice of `initial_tablets = 100` (on a single node with two cpus of 1GB memory each). This choice is not harmless - it causes 100 memtables to be created, and we run out of memory. The error messages leave **a lot** to be desired, but going down to `initial_tablets = 4` fixes this problem.
